 qustion 1.
 Let C[w, m] be the cost of transporting one truck-load of EVs from w to m.
 dicison varibale xwm denots transporting elective vehicles from warehouse 'w' to marketc[w,m], and bw denotes the number of demand from warehouse to locate,bidenotes the limit of availble stocks in warehouse.
 
minimize     c11x11+c12x12+c13x13+.......+c21x21+....+cwmxwm ,,,

 constraint  are   
 x11+x12+x13+x14.....+x1m <=b1,,

x21+x22+x23+x33+....x2m<=b2,,

...
.....

....
.........,,,

xw1+xw2+xw3+....xwm<=bw,,


 and x11+x21+x31+x41+...=b1,,
 
 x12+x22+x32+....=b2,,
 ....
 .......
 x1m+x2m+xx3m+x4m+......=bm,,


In [ ]:
!pip install -q pyomo

     |████████████████████████████████| 9.7 MB 5.1 MB/s 
     |████████████████████████████████| 49 kB 4.7 MB/s 


In [ ]:
from pyomo.environ import *

In [ ]:
import numpy as np

In [ ]:
import pandas as pd

In [ ]:
data_csvfile= pd.read_csv('lab6 ex1.csv')

In [ ]:
data_csvfile

Warehouses  Hassan  Jodhpur  Trichy  Simla  Markets Vijayawada  Dibrugarh  \
0  Ahmedabad     827      817    1470   2272                1615       3143   
1  Bengaluru     559     1832     751   3772                 637       3773   
2    Chennai    1109     2443     577   3865                 798       3521   
3      Delhi    2943     1471    4676   1337                2702       2741   
4    Kolkata    3742     3089    2931   3199                2226       1739   
5    Lucknow    2353     1160    3295    368                2369       2365   
6     Mumbai    1356     1617    2045   2941                2334       4151   
7  Hyderabad     195     1862    1331   2921                 402       2825   
8     Demand     205      300     425    270                 375        225   

   Raipur  Agartala  Avail. Stock  
0    1264      4465           350  
1    1856      5459           185  
2    1984      5303           305  
3    2482      4148           180  
4    1691      2128           205  
5    1293      3642           175  
6    1937      4287           325  
7     563      4441           370  
8     185       110          2095

In [ ]:
data_csvfile.index

RangeIndex(start=0, stop=9, step=1)

In [ ]:
print('num rows:',len(data_csvfile.index))

num rows: 9


In [ ]:
print(data_csvfile.iloc[0,:])

Warehouses            Ahmedabad
Hassan                      827
Jodhpur                     817
Trichy                     1470
Simla                      2272
Markets Vijayawada         1615
Dibrugarh                  3143
Raipur                     1264
Agartala                   4465
Avail. Stock                350
Name: 0, dtype: object


In [ ]:
print(data_csvfile.iloc[2,:])

Warehouses            Chennai
Hassan                   1109
Jodhpur                  2443
Trichy                    577
Simla                    3865
Markets Vijayawada        798
Dibrugarh                3521
Raipur                   1984
Agartala                 5303
Avail. Stock              305
Name: 2, dtype: object


In [ ]:
model=ConcreteModel()

In [ ]:
a=len(data_csvfile.index)

In [ ]:
M=a-1

In [ ]:
b=len(data_csvfile.columns)

In [ ]:
N=b-2

In [ ]:
N

8

In [ ]:
row_indices=range(M)
col_indices=range(N)

In [ ]:
print(row_indices)
print(col_indices)

range(0, 8)
range(0, 8)


In [ ]:
model=ConcreteModel()

In [ ]:
for  row_idx in row_indices:
  for col_idx in col_indices:
    print('cost[',row_idx,'][',col_idx+1,']:',data_csvfile.iloc[row_idx, col_idx+1])

cost[ 0 ][ 1 ]: 827
cost[ 0 ][ 2 ]: 817
cost[ 0 ][ 3 ]: 1470
cost[ 0 ][ 4 ]: 2272
cost[ 0 ][ 5 ]: 1615
cost[ 0 ][ 6 ]: 3143
cost[ 0 ][ 7 ]: 1264
cost[ 0 ][ 8 ]: 4465
cost[ 1 ][ 1 ]: 559
cost[ 1 ][ 2 ]: 1832
cost[ 1 ][ 3 ]: 751
cost[ 1 ][ 4 ]: 3772
cost[ 1 ][ 5 ]: 637
cost[ 1 ][ 6 ]: 3773
cost[ 1 ][ 7 ]: 1856
cost[ 1 ][ 8 ]: 5459
cost[ 2 ][ 1 ]: 1109
cost[ 2 ][ 2 ]: 2443
cost[ 2 ][ 3 ]: 577
cost[ 2 ][ 4 ]: 3865
cost[ 2 ][ 5 ]: 798
cost[ 2 ][ 6 ]: 3521
cost[ 2 ][ 7 ]: 1984
cost[ 2 ][ 8 ]: 5303
cost[ 3 ][ 1 ]: 2943
cost[ 3 ][ 2 ]: 1471
cost[ 3 ][ 3 ]: 4676
cost[ 3 ][ 4 ]: 1337
cost[ 3 ][ 5 ]: 2702
cost[ 3 ][ 6 ]: 2741
cost[ 3 ][ 7 ]: 2482
cost[ 3 ][ 8 ]: 4148
cost[ 4 ][ 1 ]: 3742
cost[ 4 ][ 2 ]: 3089
cost[ 4 ][ 3 ]: 2931
cost[ 4 ][ 4 ]: 3199
cost[ 4 ][ 5 ]: 2226
cost[ 4 ][ 6 ]: 1739
cost[ 4 ][ 7 ]: 1691
cost[ 4 ][ 8 ]: 2128
cost[ 5 ][ 1 ]: 2353
cost[ 5 ][ 2 ]: 1160
cost[ 5 ][ 3 ]: 3295
cost[ 5 ][ 4 ]: 368
cost[ 5 ][ 5 ]: 2369
cost[ 5 ][ 6 ]: 2365
cost[ 5 ][ 7 ]: 1293
cost[ 5 ][ 8 ]: 3642


In [ ]:
model.x=Var(row_indices,col_indices, domain=NonNegativeReals)

In [ ]:
model.objective = Objective(expr=sum(data_csvfile.iloc[i,j+1]*model.x[i,j] for i in row_indices for j in col_indices),sense=minimize)

In [ ]:
#model.objective=Objective(expr=sum(data_csvfile.iloc[i,j+1]*model.x[i,j] for i in row_indices for j in col_indices),sense=mimimize)

In [ ]:
model.constraint=ConstraintList()

In [ ]:
print(data_csvfile.iloc[M,1:N+1])

Hassan                205
Jodhpur               300
Trichy                425
Simla                 270
Markets Vijayawada    375
Dibrugarh             225
Raipur                185
Agartala              110
Name: 8, dtype: object


In [ ]:
for i in range(len(row_indices)):
  model.constraint.add(expr = sum(model.x[i,j] for j in col_indices)<=data_csvfile.iloc[i,N+1])

In [ ]:
for j in range(len(col_indices)):
  model.constraint.add(expr=sum(model.x[i,j] for i in row_indices) == data_csvfile.iloc[M,j+1])

In [ ]:
model.pprint()

4 Set Declarations
    constraint_index : Size=1, Index=None, Ordered=Insertion
        Key  : Dimen : Domain : Size : Members
        None :     1 :    Any :   16 : {1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16}
    x_index : Size=1, Index=None, Ordered=True
        Key  : Dimen : Domain              : Size : Members
        None :     2 : x_index_0*x_index_1 :   64 : {(0, 0), (0, 1), (0, 2), (0, 3), (0, 4), (0, 5), (0, 6), (0, 7), (1, 0), (1, 1), (1, 2), (1, 3), (1, 4), (1, 5), (1, 6), (1, 7), (2, 0), (2, 1), (2, 2), (2, 3), (2, 4), (2, 5), (2, 6), (2, 7), (3, 0), (3, 1), (3, 2), (3, 3), (3, 4), (3, 5), (3, 6), (3, 7), (4, 0), (4, 1), (4, 2), (4, 3), (4, 4), (4, 5), (4, 6), (4, 7), (5, 0), (5, 1), (5, 2), (5, 3), (5, 4), (5, 5), (5, 6), (5, 7), (6, 0), (6, 1), (6, 2), (6, 3), (6, 4), (6, 5), (6, 6), (6, 7), (7, 0), (7, 1), (7, 2), (7, 3), (7, 4), (7, 5), (7, 6), (7, 7)}
    x_index_0 : Size=1, Index=None, Ordered=Insertion
        Key  : Dimen : Domain : Size : Members
     

In [ ]:
!apt-get install -y -qq coinor-cbc

Selecting previously unselected package coinor-libcoinutils3v5.
(Reading database ... 155569 files and directories currently installed.)
Preparing to unpack .../0-coinor-libcoinutils3v5_2.10.14+repack1-1_amd64.deb ...
Unpacking coinor-libcoinutils3v5 (2.10.14+repack1-1) ...
Selecting previously unselected package coinor-libosi1v5.
Preparing to unpack .../1-coinor-libosi1v5_0.107.9+repack1-1_amd64.deb ...
Unpacking coinor-libosi1v5 (0.107.9+repack1-1) ...
Selecting previously unselected package coinor-libclp1.
Preparing to unpack .../2-coinor-libclp1_1.16.11+repack1-1_amd64.deb ...
Unpacking coinor-libclp1 (1.16.11+repack1-1) ...
Selecting previously unselected package coinor-libcgl1.
Preparing to unpack .../3-coinor-libcgl1_0.59.10+repack1-1_amd64.deb ...
Unpacking coinor-libcgl1 (0.59.10+repack1-1) ...
Selecting previously unselected package coinor-libcbc3.
Preparing to unpack .../4-coinor-libcbc3_2.9.9+repack1-1_amd64.deb ...
Unpacking coinor-libcbc3 (2.9.9+repack1-1) ...
Selecting p

In [ ]:
opt_cbc= SolverFactory('cbc',)

In [ ]:
result = opt_cbc.solve(model)
print('solver statuas:',result.solver.status)
print('solver termination condition:',result.solver.termination_condition)

solver statuas: ok
solver termination condition: optimal


In [ ]:
#displaty solution
print('\nObjective=',model.objective())
print('\n decision variable')
for i  in row_indices:
  for j in col_indices:
    print('number of the traspoted',data_csvfile['Warehouses'][i],data_csvfile.columns[j+1],':',model.x[i,j].value)
print('\nconstraint')
model.constraint.display()


Objective= 2192870.0

 decision variable
number of the traspoted Ahmedabad Hassan : 0.0
number of the traspoted Ahmedabad Jodhpur : 300.0
number of the traspoted Ahmedabad Trichy : 0.0
number of the traspoted Ahmedabad Simla : 0.0
number of the traspoted Ahmedabad Markets Vijayawada : 0.0
number of the traspoted Ahmedabad Dibrugarh : 45.0
number of the traspoted Ahmedabad Raipur : 5.0
number of the traspoted Ahmedabad Agartala : 0.0
number of the traspoted Bengaluru Hassan : 0.0
number of the traspoted Bengaluru Jodhpur : 0.0
number of the traspoted Bengaluru Trichy : 120.0
number of the traspoted Bengaluru Simla : 0.0
number of the traspoted Bengaluru Markets Vijayawada : 65.0
number of the traspoted Bengaluru Dibrugarh : 0.0
number of the traspoted Bengaluru Raipur : 0.0
number of the traspoted Bengaluru Agartala : 0.0
number of the traspoted Chennai Hassan : 0.0
number of the traspoted Chennai Jodhpur : 0.0
number of the traspoted Chennai Trichy : 305.0
number of the traspoted Chen

 7)
 the total cost are 
 2192870.0

 decision variable
number of the traspoted Ahmedabad Jodhpur : 300.0

number of the traspoted Ahmedabad Dibrugarh : 45.0
number of the traspoted Ahmedabad Raipur : 5.0

number of the traspoted Bengaluru Trichy : 120.0
number of the traspoted Bengaluru Markets Vijayawada : 65.0

number of the traspoted Chennai Trichy : 305.0

number of the traspoted Delhi Simla : 95.0
number of the traspoted Delhi Markets Vijayawada : 0.0
number of the traspoted Delhi Dibrugarh : 85.0

number of the traspoted Kolkata Dibrugarh : 95.0
number of the traspoted Kolkata Agartala : 110.0

number of the traspoted Lucknow Simla : 175.0

number of the traspoted Mumbai Hassan : 205.0

number of the traspoted Mumbai Raipur : 120.0

number of the traspoted Hyderabad Markets Vijayawada : 310.0
number of the traspoted Hyderabad Raipur : 60.0

8)
Suppose the Lucknow-Simla link is disrupted and no transportation is possible on the
Lucknow-Simla links: 
##take a large value of this , since it is maximum cost and minimisation problem ,the solution doesn't include this route
 the obje function are            True : minimize : 827*x[0,0] + 817*x[0,1] + 1470*x[0,2] + 2272*x[0,3] + 1615*x[0,4] + 3143*x[0,5] + 1264*x[0,6] + 4465*x[0,7] + 559*x[1,0] + 1832*x[1,1] + 751*x[1,2] + 3772*x[1,3] + 637*x[1,4] + 3773*x[1,5] + 1856*x[1,6] + 5459*x[1,7] + 1109*x[2,0] + 2443*x[2,1] + 577*x[2,2] + 3865*x[2,3] + 798*x[2,4] + 3521*x[2,5] + 1984*x[2,6] + 5303*x[2,7] + 2943*x[3,0] + 1471*x[3,1] + 4676*x[3,2] + 1337*x[3,3] + 2702*x[3,4] + 2741*x[3,5] + 2482*x[3,6] + 4148*x[3,7] + 3742*x[4,0] + 3089*x[4,1] + 2931*x[4,2] + 3199*x[4,3] + 2226*x[4,4] + 1739*x[4,5] + 1691*x[4,6] + 2128*x[4,7] + 2353*x[5,0] + 1160*x[5,1] + 3295*x[5,2] + 500000*x[5,3] + 2369*x[5,4] + 2365*x[5,5] + 1293*x[5,6] + 3642*x[5,7] + 1356*x[6,0] + 1617*x[6,1] + 2045*x[6,2] + 2941*x[6,3] + 2334*x[6,4] + 4151*x[6,5] + 1937*x[6,6] + 4287*x[6,7] + 195*x[7,0] + 1862*x[7,1] + 1331*x[7,2] + 2921*x[7,3] + 402*x[7,4] + 2825*x[7,5] + 563*x[7,6] + 4441*x[7,7]


In [ ]:
data_csvfile1= pd.read_csv('_lab6 ex1-2.csv')

In [ ]:
data_csvfile1

Warehouses  Hassan  Jodhpur  Trichy   Simla  Markets Vijayawada  Dibrugarh  \
0  Ahmedabad     827      817    1470    2272                1615       3143   
1  Bengaluru     559     1832     751    3772                 637       3773   
2    Chennai    1109     2443     577    3865                 798       3521   
3      Delhi    2943     1471    4676    1337                2702       2741   
4    Kolkata    3742     3089    2931    3199                2226       1739   
5    Lucknow    2353     1160    3295  500000                2369       2365   
6     Mumbai    1356     1617    2045    2941                2334       4151   
7  Hyderabad     195     1862    1331    2921                 402       2825   
8     Demand     205      300     425     270                 375        225   

   Raipur  Agartala  Avail. Stock  
0    1264      4465           350  
1    1856      5459           185  
2    1984      5303           305  
3    2482      4148           180  
4    1691      2128           205  
5    1293      3642           175  
6    1937      4287           325  
7     563      4441           370  
8     185       110          2095

In [ ]:
model.objective.deactivate()

<bound method ActiveComponentData.deactivate of <pyomo.core.base.objective.ScalarObjective object at 0x7fb4706e0980>>

In [ ]:
model.objective = Objective(expr=sum(data_csvfile1.iloc[i,j+1]*model.x[i,j] for i in row_indices for j in col_indices),sense=minimize)

This is usually indicative of a modelling error.
To avoid this warning, use block.del_component() and block.add_component().


In [ ]:
model.pprint()

4 Set Declarations
    constraint_index : Size=1, Index=None, Ordered=Insertion
        Key  : Dimen : Domain : Size : Members
        None :     1 :    Any :   16 : {1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16}
    x_index : Size=1, Index=None, Ordered=True
        Key  : Dimen : Domain              : Size : Members
        None :     2 : x_index_0*x_index_1 :   64 : {(0, 0), (0, 1), (0, 2), (0, 3), (0, 4), (0, 5), (0, 6), (0, 7), (1, 0), (1, 1), (1, 2), (1, 3), (1, 4), (1, 5), (1, 6), (1, 7), (2, 0), (2, 1), (2, 2), (2, 3), (2, 4), (2, 5), (2, 6), (2, 7), (3, 0), (3, 1), (3, 2), (3, 3), (3, 4), (3, 5), (3, 6), (3, 7), (4, 0), (4, 1), (4, 2), (4, 3), (4, 4), (4, 5), (4, 6), (4, 7), (5, 0), (5, 1), (5, 2), (5, 3), (5, 4), (5, 5), (5, 6), (5, 7), (6, 0), (6, 1), (6, 2), (6, 3), (6, 4), (6, 5), (6, 6), (6, 7), (7, 0), (7, 1), (7, 2), (7, 3), (7, 4), (7, 5), (7, 6), (7, 7)}
    x_index_0 : Size=1, Index=None, Ordered=Insertion
        Key  : Dimen : Domain : Size : Members
     

In [ ]:
!apt-get install -y -qq coinor-cbc

In [ ]:
opt_cbc= SolverFactory('cbc',)

In [ ]:
result = opt_cbc.solve(model)
print('solver statuas:',result.solver.status)
print('solver termination condition:',result.solver.termination_condition)

solver statuas: ok
solver termination condition: optimal


In [ ]:
#displaty solution
print('\nObjective=',model.objective())
print('\n decision variable')
for i  in row_indices:
  for j in col_indices:
    print('number of the traspoted',data_csvfile1['Warehouses'][i],data_csvfile1.columns[j+1],':',model.x[i,j].value)
print('\nconstraint')
model.constraint.display()


Objective= 2380570.0

 decision variable
number of the traspoted Ahmedabad Hassan : 0.0
number of the traspoted Ahmedabad Jodhpur : 300.0
number of the traspoted Ahmedabad Trichy : 0.0
number of the traspoted Ahmedabad Simla : 0.0
number of the traspoted Ahmedabad Markets Vijayawada : 0.0
number of the traspoted Ahmedabad Dibrugarh : 0.0
number of the traspoted Ahmedabad Raipur : 50.0
number of the traspoted Ahmedabad Agartala : 0.0
number of the traspoted Bengaluru Hassan : 0.0
number of the traspoted Bengaluru Jodhpur : 0.0
number of the traspoted Bengaluru Trichy : 120.0
number of the traspoted Bengaluru Simla : 0.0
number of the traspoted Bengaluru Markets Vijayawada : 65.0
number of the traspoted Bengaluru Dibrugarh : 0.0
number of the traspoted Bengaluru Raipur : 0.0
number of the traspoted Bengaluru Agartala : 0.0
number of the traspoted Chennai Hassan : 0.0
number of the traspoted Chennai Jodhpur : 0.0
number of the traspoted Chennai Trichy : 305.0
number of the traspoted Chen

9)the tolat cost are 2380570.0

 decision variable
number of the traspoted Ahmedabad Jodhpur : 300.0

number of the traspoted Ahmedabad Raipur : 50.0
number of the traspoted Bengaluru Trichy : 120.0
number of the traspoted Bengaluru Markets Vijayawada : 65.0

number of the traspoted Chennai Trichy : 305.0


number of the traspoted Delhi Simla : 180.0


number of the traspoted Kolkata Dibrugarh : 95.0
number of the traspoted Kolkata Agartala : 110.0

number of the traspoted Lucknow Markets Vijayawada : 0.0
number of the traspoted Lucknow Dibrugarh : 130.0
number of the traspoted Lucknow Raipur : 45.0

number of the traspoted Mumbai Hassan : 205.0
0
number of the traspoted Mumbai Simla : 90.0

number of the traspoted Mumbai Raipur : 30.0

number of the traspoted Hyderabad Markets Vijayawada : 310.0

number of the traspoted Hyderabad Raipur : 60.0
comment:the cost is incresing when e the Lucknow-Simla link is disrupted and no transportation the solution are exist but this solution are   not optmial solution

10)Suppose in addition to Lucknow-Simla link, the Kolkata-Dibrugarh link and Delhi-Raipur
link are disrupted and no transportation is possible along these links as well:

take a large value  all of  this route , since it is maximum cost and minimisation problem ,the solution doesn't include this 
 now new obje function are
          True : minimize : 827*x[0,0] + 817*x[0,1] + 1470*x[0,2] + 2272*x[0,3] + 1615*x[0,4] + 3143*x[0,5] + 1264*x[0,6] + 4465*x[0,7] + 559*x[1,0] + 1832*x[1,1] + 751*x[1,2] + 3772*x[1,3] + 637*x[1,4] + 3773*x[1,5] + 1856*x[1,6] + 5459*x[1,7] + 1109*x[2,0] + 2443*x[2,1] + 577*x[2,2] + 3865*x[2,3] + 798*x[2,4] + 3521*x[2,5] + 1984*x[2,6] + 5303*x[2,7] + 2943*x[3,0] + 1471*x[3,1] + 4676*x[3,2] + 1337*x[3,3] + 2702*x[3,4] + 2741*x[3,5] + 700000*x[3,6] + 4148*x[3,7] + 3742*x[4,0] + 3089*x[4,1] + 2931*x[4,2] + 3199*x[4,3] + 2226*x[4,4] + 600000*x[4,5] + 1691*x[4,6] + 2128*x[4,7] + 2353*x[5,0] + 1160*x[5,1] + 3295*x[5,2] + 500000*x[5,3] + 2369*x[5,4] + 2365*x[5,5] + 1293*x[5,6] + 3642*x[5,7] + 1356*x[6,0] + 1617*x[6,1] + 2045*x[6,2] + 2941*x[6,3] + 2334*x[6,4] + 4151*x[6,5] + 1937*x[6,6] + 4287*x[6,7] + 195*x[7,0] + 1862*x[7,1] + 1331*x[7,2] + 2921*x[7,3] + 402*x[7,4] + 2825*x[7,5] + 563*x[7,6] + 4441*x[7,7]


In [ ]:

data_csvfile2= pd.read_csv('_lab6 ex1-3.csv')

In [ ]:
model.objective.deactivate()

<bound method ActiveComponentData.deactivate of <pyomo.core.base.objective.ScalarObjective object at 0x7fb478e96360>>

In [ ]:
model.objective = Objective(expr=sum(data_csvfile2.iloc[i,j+1]*model.x[i,j] for i in row_indices for j in col_indices),sense=minimize)

This is usually indicative of a modelling error.
To avoid this warning, use block.del_component() and block.add_component().


In [ ]:
model.pprint()

4 Set Declarations
    constraint_index : Size=1, Index=None, Ordered=Insertion
        Key  : Dimen : Domain : Size : Members
        None :     1 :    Any :   16 : {1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16}
    x_index : Size=1, Index=None, Ordered=True
        Key  : Dimen : Domain              : Size : Members
        None :     2 : x_index_0*x_index_1 :   64 : {(0, 0), (0, 1), (0, 2), (0, 3), (0, 4), (0, 5), (0, 6), (0, 7), (1, 0), (1, 1), (1, 2), (1, 3), (1, 4), (1, 5), (1, 6), (1, 7), (2, 0), (2, 1), (2, 2), (2, 3), (2, 4), (2, 5), (2, 6), (2, 7), (3, 0), (3, 1), (3, 2), (3, 3), (3, 4), (3, 5), (3, 6), (3, 7), (4, 0), (4, 1), (4, 2), (4, 3), (4, 4), (4, 5), (4, 6), (4, 7), (5, 0), (5, 1), (5, 2), (5, 3), (5, 4), (5, 5), (5, 6), (5, 7), (6, 0), (6, 1), (6, 2), (6, 3), (6, 4), (6, 5), (6, 6), (6, 7), (7, 0), (7, 1), (7, 2), (7, 3), (7, 4), (7, 5), (7, 6), (7, 7)}
    x_index_0 : Size=1, Index=None, Ordered=Insertion
        Key  : Dimen : Domain : Size : Members
     

In [ ]:
!apt-get install -y -qq coinor-cbc

In [ ]:
opt_cbc= SolverFactory('cbc',)

In [ ]:
result = opt_cbc.solve(model)
print('solver statuas:',result.solver.status)
print('solver termination condition:',result.solver.termination_condition)

solver statuas: ok
solver termination condition: optimal


In [ ]:
#displaty solution
print('\nObjective=',model.objective())
print('\n decision variable')
for i  in row_indices:
  for j in col_indices:
    print('number of the traspoted',data_csvfile2['Warehouses'][i],data_csvfile2.columns[j+1],':',model.x[i,j].value)
print('\nconstraint')
model.constraint.display()


Objective= 2518200.0

 decision variable
number of the traspoted Ahmedabad Hassan : 0.0
number of the traspoted Ahmedabad Jodhpur : 300.0
number of the traspoted Ahmedabad Trichy : 0.0
number of the traspoted Ahmedabad Simla : 0.0
number of the traspoted Ahmedabad Markets Vijayawada : 0.0
number of the traspoted Ahmedabad Dibrugarh : 50.0
number of the traspoted Ahmedabad Raipur : 0.0
number of the traspoted Ahmedabad Agartala : 0.0
number of the traspoted Bengaluru Hassan : 0.0
number of the traspoted Bengaluru Jodhpur : 0.0
number of the traspoted Bengaluru Trichy : 120.0
number of the traspoted Bengaluru Simla : 0.0
number of the traspoted Bengaluru Markets Vijayawada : 65.0
number of the traspoted Bengaluru Dibrugarh : 0.0
number of the traspoted Bengaluru Raipur : 0.0
number of the traspoted Bengaluru Agartala : 0.0
number of the traspoted Chennai Hassan : 0.0
number of the traspoted Chennai Jodhpur : 0.0
number of the traspoted Chennai Trichy : 305.0
number of the traspoted Chen

11)
the tolat cost 2518200.0

 decision variable
number of the traspoted Ahmedabad Jodhpur : 300.0

number of the traspoted Ahmedabad Dibrugarh : 50.0

number of the traspoted Bengaluru Trichy : 120.0

number of the traspoted Chennai Trichy : 305.0
0
number of the traspoted Delhi Simla : 180.0


number of the traspoted Kolkata Raipur : 95.0
number of the traspoted Kolkata Agartala : 110.0

number of the traspoted Lucknow Dibrugarh : 175.0

number of the traspoted Mumbai Hassan : 205.0

number of the traspoted Mumbai Simla : 90.0

number of the traspoted Mumbai Raipur : 30.0

number of the traspoted Hyderabad Markets Vijayawada : 310.0
number of the traspoted Hyderabad Raipur : 60.0
,

objervation:Lucknow-Simla link, the Kolkata-Dibrugarh link and Delhi-Raipur
link are disrupted and no transportation is possible along these links as well,
 the cost are incerseing , then this is not optimal solution